gmso/external/convert_networkx.py

```diff
@@ -0,0 +1,84 @@
+import networkx as nx
+
+from gmso.exceptions import GMSOError
+from gmso.abc.abstract_connection import Connection
+from gmso.core.bond import Bond
+from gmso.abc.abstract_site import Site
+from gmso.core.topology import Topology
+
+
+def from_networkx(graph):
+    """Convert a networkx.Graph to a gmso.Topology
+
+    Creates a topology from the graph where each node is a site and each
+    edge becomes a connection.
+
+    Parameters
+    ----------
+    graph : networkX.Graph
+        networkx.Graph instance that need to be converted
+
+    Returns
+    -------
+    top : gmso.Topology
+
+    Notes
+    -----
+    - While a lot of information is lost from converting to a graph object
+    (e.g. metadata, mixing rules, etc.), the graph representation is a
+    useful way to manipulate and extract connectivity information from
+    Topology objects.
+    - The edge has a `connection` attribute, which stores the Bond
+    object it was created from
+    """
+
```

Comment:
I think we should have a validation check here to either ensure it is a `networkx` graph object, or try to coerce into a `networkx` graph